cli/src/commands/explore/calls.rs

```diff
@@ -0,0 +1,145 @@
+use crate::utils::type_description::print_type_description;
+use crate::utils::type_example::print_type_examples;
+use crate::utils::with_indent;
+use clap::Args;
+
+use std::fmt::Write;
+use std::write;
+
+use color_eyre::eyre::eyre;
+use frame_metadata::v15::PalletMetadata;
+
+use scale_info::form::PortableForm;
+use scale_info::{PortableRegistry, Type, TypeDef, TypeDefVariant};
+use scale_value::{Composite, ValueDef};
+
+use subxt::tx;
+use subxt::utils::H256;
+use subxt::{config::SubstrateConfig, Metadata, OfflineClient};
+
+#[derive(Debug, Clone, Args)]
+pub struct CallsSubcommand {
+    call: Option<String>,
+    #[clap(required = false)]
+    trailing_args: Vec<String>,
+}
+
+pub(crate) fn explore_calls(
+    command: CallsSubcommand,
+    metadata: &Metadata,
+    pallet_metadata: &PalletMetadata<PortableForm>,
+) -> color_eyre::Result<()> {
+    let pallet_name = pallet_metadata.name.as_str();
+
+    // get the enum that stores the possible calls:
+    let (calls_enum_type_def, _calls_enum_type) =
+        get_calls_enum_type(pallet_metadata, &metadata.runtime_metadata().types)?;
+
+    // if no call specified, show user the calls to choose from:
+    let Some(call_name) = command.call else {
+        let available_calls = print_available_calls(calls_enum_type_def, pallet_name);
+        println!("Usage:\n    subxt explore {pallet_name} calls <CALL>\n        explore a specific call within this pallet\n\n{available_calls}", );
+        return Ok(());
+    };
+
+    // if specified call is wrong, show user the calls to choose from (but this time as an error):
+    let Some(call) = calls_enum_type_def.variants.iter().find(|variant| variant.name.to_lowercase() == call_name.to_lowercase())   else {
```

Comment:
```suggestion
    let Some(call) = calls_enum_type_def.variants.iter().find(|variant| variant.name.to_lowercase() == call_name.to_lowercase()) else {
```

Reply:
Hmm I wonder why rustfmt does not catch these cases.